cmd/thanos/bucket.go

```diff
@@ -213,4 +217,191 @@ func registerBucket(m map[string]setupFunc, app *kingpin.Application, name strin
 			return printBlock(id)
 		})
 	}
+
+	inspect := cmd.Command("inspect", "inspect all blocks in the bucket")
+	selector := inspect.Flag("selector ", "Selects blocks based on label, e.g. -l key1=value1,key2=value2. All key value pairs must match.").Short('l').
+		Default("").String()
+	sortBy := inspect.Flag("sort-by", "Sort by columns.").
+		Default("FROM,UNTIL").String()
+
+	m[name+" inspect"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer, _ bool) error {
+
+		// Parse selector.
+		selectorMap := make(map[string]string)
+		selectorsSplit := strings.Split(*selector, ",")
```

Comment:
wonder if you cannot use `cmd/thanos/rule.go:795` function:
```
func parseFlagLabels(s []string) (labels.Labels, error) {
	var lset labels.Labels
	for _, l := range s {
		parts := strings.SplitN(l, "=", 2)
		if len(parts) != 2 {
			return nil, errors.Errorf("unrecognized label %q", l)
		}
		val, err := strconv.Unquote(parts[1])
		if err != nil {
			return nil, errors.Wrap(err, "unquote label value")
		}
		lset = append(lset, labels.Label{Name: parts[0], Value: val})
	}
	return lset, nil
}
```

Reasons:
* Less code (reuse)
* labels.Labels type have nice equal, compare capabilities built in.

Reply:
We could actually introduce query like matching, but that might be overkill, sticking to labels.Labels is enough (:

Reply:
I changed the code accordingly. This has the side effect  that the values must be quoted now in the selector (-l name="value"), but I think that's okay, especially because it's consistent with PromQL.

Unfortunately, I'm not able to use the equal function. What I actually need is a subset function :). That's because not all block labels must be explicitly listed in the selector. It's enough if the selector labels are a subset of the block label.s